### PR TITLE
use setup-r-dependencies action

### PR DESCRIPTION
**Pull request by pawelru**
- update Quarto extensions used
- use setup-r-dependencies

companion PRs:
- https://github.com/insightsengineering/r.pkg.template/pull/260
- https://github.com/insightsengineering/setup-r-dependencies/pull/14
- https://github.com/insightsengineering/setup-r-dependencies/pull/15

### CODE REVIEW

File: .github/workflows/check.yml [hunk 1]
```diff
@@ -20,34 +20,59 @@ jobs:
     if: github.event_name != 'push'
     name: Audit Dependencies 🕵️‍♂️
     uses: insightsengineering/r.pkg.template/.github/workflows/audit.yaml@main
-  r-cmd:
+  r-cmd-dev:
     if: github.event_name != 'push'
     name: R CMD Check (development) 🧬
-    uses: insightsengineering/r.pkg.template/.github/workflows/build-check-install.yaml@main
+    uses: insightsengineering/r.pkg.template/.github/workflows/build-check-install.yaml@bci_params
```
Comment (pawelru) [inline .github/workflows/check.yml:26]:
```suggestion
    uses: insightsengineering/r.pkg.template/.github/workflows/build-check-install.yaml@main
```

File: .github/workflows/docs.yml [hunk 2]
```diff
@@ -70,14 +70,15 @@ jobs:
           echo "gchat_webhook=${{ secrets.GCHAT_WEBHOOK }}" >> $GITHUB_ENV
         shell: bash
 
-      - name: Run Staged dependencies 🎦
-        uses: insightsengineering/staged-dependencies-action@v1
-        env:
-          GITHUB_PAT: ${{ secrets.REPO_GITHUB_TOKEN }}
+      - name: Setup R dependencies 🎦
+        uses: insightsengineering/setup-r-dependencies@named_repos
```
Comment (pawelru) [inline .github/workflows/docs.yml:74]:
```suggestion
        uses: insightsengineering/setup-r-dependencies@v1
```

File: .github/workflows/docs.yml [hunk 3]
```diff
@@ -16,6 +16,7 @@ on:
   push:
     branches:
       - main
+      - use_setup_r_deps
```
Comment (pawelru) [inline .github/workflows/docs.yml:19]:
```suggestion
```

File: .github/workflows/docs.yml [hunk 4]
```diff
@@ -112,28 +113,28 @@ jobs:
...
-        with:
-          github_token: ${{ secrets.GITHUB_TOKEN }}
-          publish_dir: ./book/_site
-          destination_dir: development
+      #- name: Publish docs 📔
+      #  uses: peaceiris/actions-gh-pages@v3
+      #  with:
+      #    github_token: ${{ secrets.GITHUB_TOKEN }}
+      #    publish_dir: ./book/_site
+      #    destination_dir: development
```
Comment (pawelru) [inline .github/workflows/docs.yml:129]:
```suggestion
      - name: Publish docs 📔
        uses: peaceiris/actions-gh-pages@v3
        with:
          github_token: ${{ secrets.GITHUB_TOKEN }}
          publish_dir: ./book/_site
          destination_dir: development
```

File: .github/workflows/docs.yml [hunk 5]
```diff
@@ -112,28 +113,28 @@ jobs:
...
-          name: ${{ github.event.repository.name }} - development
-          url: ${{ secrets.GCHAT_WEBHOOK }}
-          status: ${{ job.status }}
+      #- name: GChat notification 🔔
+      #  if: (failure() || cancelled()) && env.gchat_webhook != ''
+      #  uses: insightsengineering/google-chat-notification@master
+      #  with:
+      #    name: ${{ github.event.repository.name }} - development
+      #    url: ${{ secrets.GCHAT_WEBHOOK }}
+      #    status: ${{ job.status }}
```
Comment (pawelru) [inline .github/workflows/docs.yml:137]:
```suggestion
      - name: GChat notification 🔔
        if: (failure() || cancelled()) && env.gchat_webhook != ''
        uses: insightsengineering/google-chat-notification@master
        with:
          name: ${{ github.event.repository.name }} - development
          url: ${{ secrets.GCHAT_WEBHOOK }}
          status: ${{ job.status }}
```

File: .github/workflows/docs.yml [hunk 6]
```diff
@@ -193,36 +198,35 @@ jobs:
...
-        with:
-          github_token: ${{ secrets.GITHUB_TOKEN }}
-          publish_dir: ./book/_site
-          destination_dir: stable
+      #- name: Publish docs 📔
+      #  uses: peaceiris/actions-gh-pages@v3
+      #  with:
+      #    github_token: ${{ secrets.GITHUB_TOKEN }}
+      #    publish_dir: ./book/_site
+      #    destination_dir: stable
```
Comment (pawelru) [inline .github/workflows/docs.yml:221]:
```suggestion
      - name: Publish docs 📔
        uses: peaceiris/actions-gh-pages@v3
        with:
          github_token: ${{ secrets.GITHUB_TOKEN }}
          publish_dir: ./book/_site
          destination_dir: stable
```

File: .github/workflows/docs.yml [hunk 7]
```diff
@@ -193,36 +198,35 @@ jobs:
...
-          name: ${{ github.event.repository.name }} - stable
-          url: ${{ secrets.GCHAT_WEBHOOK }}
-          status: ${{ job.status }}
+      #- name: GChat notification 🔔
+      #  if: (failure() || cancelled()) && env.gchat_webhook != ''
+      #  uses: insightsengineering/google-chat-notification@master
+      #  with:
+      #    name: ${{ github.event.repository.name }} - stable
+      #    url: ${{ secrets.GCHAT_WEBHOOK }}
+      #    status: ${{ job.status }}
```
Comment (pawelru) [inline .github/workflows/docs.yml:229]:
```suggestion
      - name: GChat notification 🔔
        if: (failure() || cancelled()) && env.gchat_webhook != ''
        uses: insightsengineering/google-chat-notification@master
        with:
          name: ${{ github.event.repository.name }} - stable
          url: ${{ secrets.GCHAT_WEBHOOK }}
          status: ${{ job.status }}
```

File: .github/workflows/docs.yml [hunk 8]
```diff
@@ -161,11 +162,15 @@ jobs:
           echo "gchat_webhook=${{ secrets.GCHAT_WEBHOOK }}" >> $GITHUB_ENV
         shell: bash
 
-      - name: Install packages 🎦
-        run: |
-          devtools::install_dev_deps(".", repos = c("https://insightsengineering.r-universe.dev/", "https://cloud.r-project.org/"))
-        shell: Rscript {0}
-        working-directory: package
+      - name: Setup R dependencies 🎦
+        uses: insightsengineering/setup-r-dependencies@named_repos
```
Comment (pawelru) [inline .github/workflows/docs.yml:166]:
```suggestion
        uses: insightsengineering/setup-r-dependencies@v1
```

File: .github/workflows/check.yml [hunk 9]
```diff
@@ -20,34 +20,59 @@ jobs:
...
+        insightsengineering/teal.logger
+        insightsengineering/teal.reporter
+        insightsengineering/teal.slice
+        insightsengineering/teal.transform
+        insightsengineering/teal.widgets
+        insightsengineering/teal
+        insightsengineering/teal.modules.general
+        insightsengineering/teal.modules.clinical
+      skip-desc-dev: true
+      repository-list: "https://pharmaverse.r-universe.dev/, PPM@latest"
```
Comment (pawelru) [inline .github/workflows/check.yml:56]:
```suggestion
      repository-list: "r-universe=https://pharmaverse.r-universe.dev/, PPM=PPM@latest"
```

File: .github/workflows/check.yml [hunk 10]
```diff
@@ -20,34 +20,59 @@ jobs:
...
       package-subdirectory: package
       additional-env-vars: |
         NOT_CRAN=true
         QUARTO_PROFILE=stable
         TLG_CATALOG_PKG_BUILD_RENDER=TRUE
       concurrency-group: stable
+      deps-installation-method: setup-r-dependencies
+      skip-desc-dev: true
+      skip-desc-branch: true
+      repository-list: "https://insightsengineering.r-universe.dev/, PPM@latest"
```
Comment (pawelru) [inline .github/workflows/check.yml:74]:
```suggestion
      repository-list: "r-universe-stable=https://insightsengineering.r-universe.dev/, PPM=PPM@latest"
```

File: .github/workflows/docs.yml [hunk 11]
```diff
@@ -70,14 +69,15 @@ jobs:
...
+        uses: insightsengineering/setup-r-dependencies@v1
         with:
-          path: "./package"
-          enable-check: false
-          direction: upstream
+          github-token: ${{ secrets.REPO_GITHUB_TOKEN }}
+          repository-path: "./package"
+          skip-desc-branch: true
+          skip-desc-dev: true
+          repository-list: "https://pharmaverse.r-universe.dev/, PPM@latest"
```
Comment (pawelru) [inline .github/workflows/docs.yml:79]:
```suggestion
          repository-list: "r-universe-dev=https://pharmaverse.r-universe.dev/, PPM=PPM@latest"
```

File: .github/workflows/docs.yml [hunk 12]
```diff
@@ -161,11 +161,15 @@ jobs:
...
-        shell: Rscript {0}
-        working-directory: package
+      - name: Setup R dependencies 🎦
+        uses: insightsengineering/setup-r-dependencies@v1
+        with:
+          github-token: ${{ secrets.REPO_GITHUB_TOKEN }}
+          repository-path: "./package"
+          skip-desc-branch: true
+          skip-desc-dev: true
+          repository-list: "https://insightsengineering.r-universe.dev/, PPM@latest"
```
Comment (pawelru) [inline .github/workflows/docs.yml:171]:
```suggestion
          repository-list: "r-universe-stable=https://insightsengineering.r-universe.dev/, PPM=PPM@latest"
```

**Comment by m7pr** [comment]
Hey @pawelru is this ready for a review? R CMD Check is failiing

**Comment by pawelru** [comment]
Yes it is. R CMD CHECK error is unrelated and should be handled separately. I will create a separate PR and there will ask SME team for a review. Here, I want to ask for a review of changes in the actions definition

File: .github/workflows/check.yml [hunk 13]
```diff
@@ -20,34 +20,59 @@ jobs:
...
+        insightsengineering/teal.code
+        insightsengineering/teal.logger
+        insightsengineering/teal.reporter
+        insightsengineering/teal.slice
+        insightsengineering/teal.transform
+        insightsengineering/teal.widgets
+        insightsengineering/teal
+        insightsengineering/teal.modules.general
+        insightsengineering/teal.modules.clinical
+      skip-desc-dev: true
```
Comment (m7pr) [inline .github/workflows/check.yml:55]: @pawelru what is the purpose of this parameter?

File: .github/workflows/docs.yml [hunk 14]
```diff
@@ -70,14 +69,15 @@ jobs:
...
-        uses: insightsengineering/staged-dependencies-action@v1
-        env:
-          GITHUB_PAT: ${{ secrets.REPO_GITHUB_TOKEN }}
+      - name: Setup R dependencies 🎦
+        uses: insightsengineering/setup-r-dependencies@v1
         with:
-          path: "./package"
-          enable-check: false
-          direction: upstream
+          github-token: ${{ secrets.REPO_GITHUB_TOKEN }}
```
Comment (m7pr) [inline .github/workflows/docs.yml:75]: Why not GITHUB_PAT env ? as in here https://github.com/insightsengineering/r.pkg.template/pull/262 ?

File: .github/workflows/check.yml (hunk 13 above)
Comment (m7pr) [reply .github/workflows/check.yml:55]: Ah, I see this is passed in here
https://github.com/insightsengineering/r.pkg.template/pull/260/files#diff-4db3492d6468586dd50add2ee731b86d3211cad713aa03e56c7d059d891a0cdaR147-R174

Comment (pawelru) [reply .github/workflows/check.yml:55]: Yes. It i passed to the `setup-r-dependencies` and utilised there. 
it has to be analysed in the context of the following parameter - `repository-list`. This points to the r-universe with universe of our devel packages. Hence we can skip the logic of identifying dev deps as this is essentially redundant.

File: .github/workflows/docs.yml (hunk 14 above)
Comment (pawelru) [reply .github/workflows/docs.yml:75]: not yet merged but you got the point. Will change after the merge

**Comment by m7pr** [comment]
Pawel, I think I understand changes in YAML/Actions part.
We move to setup-r-dependencies installation strategy, that recently got few new parameters. This is understandable and OK.

I just can't help with the JS part that is changed in this PR. Would you like some other pair of eyes?

File: .github/workflows/check.yml (hunk 13 above)
Comment (m7pr) [reply .github/workflows/check.yml:55]: Amazing, thanks for the reply

File: .github/workflows/docs.yml (hunk 14 above)
Comment (m7pr) [reply .github/workflows/docs.yml:75]: Alrighty!

**Comment by pawelru** [comment]
> I just can't help with the JS part that is changed in this PR. Would you like some other pair of eyes?

These are actually external Quarto extension code _imported_ (included) in here. That's how Quarto handles this - [docs](https://quarto.org/docs/extensions/managing.html). I just updated this.

**Review by m7pr — APPROVED** [review]
Besides automated Quarto edits, no changes needed. Looks good.
Glad that we are able to use dev and stable R-universe in setup-r-dependencies

**Pull request merged**